### ISSUE TITLE
PHP 8.1

### ISSUE THREAD

**Issue by toitzi** [koseven/koseven#471]
Reported, verified and already known from Telegram, i just opend the issue so we do not loose track.

![image](https://user-images.githubusercontent.com/42447585/162963700-7512fd0c-46dd-4cda-bb5b-0f89cfb17bd5.png)
![image](https://user-images.githubusercontent.com/42447585/162963732-f6ea51f9-9e2a-4f1a-a2d5-4761137b1bda.png)

**Comment by tomvano** [comment]
I've been using this https://github.com/piotrbaczek/koseven/tree/feature/cookie-same-site in production for a while. Works like a charm, but it could break compatibility.

**Comment by tomvano** [comment]
My PRs should fix the issues mentioned above.
If you need help with anything else on the master branch, let me know.

**Comment by CasperBE** [comment]
Would it be possible to tag a new version that is PHP 8.1 compatible?
Or are there still any issues that need to be solved?

**Comment by saruman** [comment]
Yes, some issues in 8.1.  Im fixing the issues myself, also creating pull requests.

**Comment by CasperBE** [comment]
I've added some PR's myself, I will add more when I find more issues.

**Comment by Daijobou** [comment]
Maybe it would be good to become compatible with PHP 8.2, not only PHP 8.1? Has anyone already looked at this?  I found this: https://github.com/koseven/koseven/blob/master/system/utf8/strlen.php#L16

See PHP 8.2: utf8_encode and utf8_decode functions deprecated:
https://php.watch/versions/8.2/utf8_encode-utf8_decode-deprecated
https://stitcher.io/blog/new-in-php-82#utf8_encode()-and-utf8_decode()-deprecations-rfc

and maybe https://stitcher.io/blog/new-in-php-82#locale-insensitive-strtolower()-and-strtoupper()-breaking-rfc

**Comment by jstrobel** [comment]
@Daijobou Sorry myself have no experience with koseven on PHP 8.2. It think it would be good if you ask this question in the telegram-group where a lot more people follow.

**Comment by phpclub** [comment]
Unreal, bro @toitzi @jstrobel 
Under PHP 8.3
ERRORS!
Tests: 1472, Assertions: 2767, Errors: 99, Failures: 2, Warnings: 10, Skipped: 90, Risky: 1.

Trt to fix
  "require-dev": {
    "phpunit/phpunit": "^9.3.11",
    "phpunit/phpcov": "^8.2.1",
    "php-coveralls/php-coveralls": "^2.0"
  },


https://github.com/sebastianbergmann/phpunit/issues/3338
https://github.com/sebastianbergmann/phpunit/issues/3368

**Comment by phpclub** [comment]
A separate branch will be needed - to work in PHP 8 
@jstrobel @toitzi

**Comment by toitzi** [comment]
@phpclub yes, there is a lot more then i thought, however im currently down to 1 Failed test, it seems like we need to get rid of "internations/http-mock" as dependency and handle this differently. (Sodium and redis tests excluded for now)

**Comment by phpclub** [comment]
I'm trying this  php configuration - work fine 
 7.3 & php 8.3 together 
"require": {
    "php": ">=7.2"
  },
  "require-dev": {
    "phpunit/phpunit": "^9.3.11",
    "phpunit/phpcov": "^8.2.1",
    "php-coveralls/php-coveralls": "^2.0",
    "phpunit/phpunit-dom-assertions": "^2.6"
  },

**Comment by phpclub** [comment]
Try php 8.3 workaround like
if ( version_compare(PHP_VERSION, '8.3.0', '>=') ) {
    try {
      $res = mb_str_pad($str, $final_str_length, $pad_str, $pad_type);
      return $res;
    } catch  (Exception $e) {
      throw new UTF8_Exception(
        "UTF8::str_pad: Unknown padding type (:pad_type)",
        [':pad_type' => $pad_type]);
    }
  } else {

**Comment by Neustradamus** [comment]
Any progress on it?

**Comment by uBizzy** [comment]
> Any progress on it?

The php8.3 branch is usable right now. Now ready for production because it was not testes enough, but it's usable and most of the errors were fixed. If you are willing to help, use it and fix some issues or open an issue. If more people start using it, more we will improve it.

**Comment by Neustradamus** [comment]
Hope a new version with support.

The latest is 3.3.10 (2020-10-12):
- https://github.com/koseven/koseven/releases

On https://github.com/koseven:
Please update "Koseven a Kohana fork compatible with PHP7"

On https://github.com/koseven/koseven:
Please update "Koseven a Kohana fork compatible with PHP7"

It is not only PHP 7.x.

**Comment by tomvano** [comment]
The default for KO7_Cookie::$domain should be '' instead of null

**Comment by adminvl-it** [comment]
Solved the problem, I have the same, until I realized how to solve it, re-read the thread.

What do I need to fix?

```
		catch (Exception $e)
			{
			// Error reading the session, usually a corrupt session.
			throw new Session_Exception('Error reading session data.', NULL, Session_Exception::SESSION_CORRUPT);
			}
```

```
	protected static function _setcookie ($name, $value, $expire, $path, $domain, $secure, $httponly)
		{
		return setcookie ($name, $value ?? '', $expire, $path, $domain ?? '', $secure, $httponly);
		}
```

![Image](https://github.com/user-attachments/assets/92f95583-9bb9-4f44-80ed-d740b3403a19)

**Comment by augmenter** [comment]
anyone notice that new Model_Test(Array('field'=>'value')) does not return a result?

**Comment by pronego-info** [comment]
Any updates on PHP 8.3 compatibility? We’re also following this closely at koeight.dev.

**Comment by Neustradamus** [comment]
Important PHP 8.3 development will be stopped soon and PHP 8.4 exists since 2024-11-21 and PHP 8.5.0 Alpha 4 since 2025-08-01, Beta 1 tomorrow 2025-08-14:
- https://www.php.net/archive/2025.php#2025-07-31-3
- https://github.com/php/php-src/releases

Linked to:
- https://github.com/koseven/koseven/issues/543
- https://github.com/koseven/koseven/issues/535
- https://github.com/koseven/koseven/issues/471

**Comment by Daijobou** [comment]
Your request seems contradictory. On your website you offer to migrate Koseven projects to newer PHP versions for a fee, yet here you are asking about PHP 8.3 compatibility. I have been running Koseven with PHP 8.4 for some time without issues for the features I use, and the necessary adjustments are already available in the repository and community. If you are selling migration services you should be aware of and use these existing solutions, otherwise it looks like you are profiting from the work of others.

**Comment by Neustradamus** [comment]
@toitzi: Can you rename this title to "PHP 8.3.x support"?

Recall: Please note that PHP 8.3 development will be stopped soon and there is now PHP 8.4 (since November 2024) and PHP 8.5 Beta 1 will be released tomorrow (2025-08-14) after several Alpha versions and before PHP 9.

Linked to tickets:
- https://github.com/koseven/koseven/issues/471
- https://github.com/koseven/koseven/issues/535
- https://github.com/koseven/koseven/issues/543

Linked to PRs:
- https://github.com/koseven/koseven/pull/522
- https://github.com/koseven/koseven/pull/527
- https://github.com/koseven/koseven/pull/528
- https://github.com/koseven/koseven/pull/531
- https://github.com/koseven/koseven/pull/538
- https://github.com/koseven/koseven/pull/542

**Comment by timhj** [comment]
There are still a number of issues on the `php8.3` tree which stop it being a usable version for production. The design decision to move from `Koseven::` to `KO7::` also creates difficulty comparing historical code with updates and what needs to be fixed.

@toitzi Our team is happy to help where we can, just let us know what's needed.
